mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java

```diff
@@ -175,6 +186,47 @@ private void setInstanceIdToRecord(Record record, JsonObject instance) {
     record.getExternalIdsHolder().setInstanceId(instanceId);
   }
 
+  private void updateLatestTransactionDate(Record record, HashMap<String, String> context) throws JsonProcessingException {
+    if (isField005NeedToUpdate(record, context)) {
+      String date = AdditionalFieldsUtil.dateTime005Formatter.format(ZonedDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()));
+      boolean isLatestTransactionDateUpdated = AdditionalFieldsUtil.addControlledFieldToMarcRecord(record, AdditionalFieldsUtil.TAG_005, date, true);
+      if (!isLatestTransactionDateUpdated) {
+        throw new PostProcessingException(format("Failed to update field '005' to record with id '%s'", record.getId()));
+      }
+    }
+  }
+
+  private boolean isField005NeedToUpdate(Record record, HashMap<String, String> context) throws JsonProcessingException {
```

Comment:
Plz move all field methods to the additional field util

Reply:
Fixed.